Bottom: 6e0fd8ead10120e20da2e19327260cbd849459b9
Top:    e38929de3220f387d13ed5d7cbf3e9beacbd0cf3
Author: Parth Shah <parth@linux.ibm.com>
Date:   2020-02-12 15:00:45 +0530

Refresh of sched-fair-allow-pre-empted

---

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index f3347eced9f5..07abb608dd0e 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -5926,7 +5926,7 @@ static int select_idle_cpu(struct task_struct *p, struct sched_domain *sd, int t
 	delta = (s64)(time - cost) / 8;
 	this_sd->avg_scan_cost += delta;
 
-	return cpi;
+	return cpu != -1 ? cpu : cpi;
 }
 
 /*
